api/v1alpha3/virtualmachine_types.go

```diff
@@ -406,6 +406,13 @@ type VirtualMachineSpec struct {
 	// behavior. In other words, please be careful when choosing to upgrade a
 	// VM to a newer hardware version.
 	MinHardwareVersion int32 `json:"minHardwareVersion,omitempty"`
+
+	// BiosUUID describes the desired BIOS UUID for a VM.
+	// If omitted, this field defaults to a random UUID.
+	// When the bootstrap provider is Cloud-Init, this value is
+	// used as the Cloud-Init instance ID.
+	// +optional
+	BiosUUID string `json:"biosUUID,omitempty"`
```

Comment:
This should also validate the format, ex.

```go
// +kubebuilder:validation:Format:=uuid4
```

See https://github.com/kubernetes/kubernetes/blob/ac8ac0fc170eec7cdf292d841f8d026a65ce1681/vendor/github.com/go-openapi/strfmt/format.go#L96-L169 for a list of not super well-known formats that can be used.

Reply:
Nice, will do.